OFEP-metric-hooks.md

```diff
@@ -0,0 +1,112 @@
+metrics. Providing telemetry data out of the box will make OpenFeature attractive for both users and vendors.
+
+## Proposal
+
+The proposal here is to define a set of metrics that can be used with different hook stages. For example, error metrics
+can be collected at the hook's `error` stage. Going with this background, I propose the following metrics.
+
+## Metrics
+
+All metrics defined through this proposal carry OpenTelemetry semantic conventions defined attributes(dimensions)[1].
+
+- feature_flag.key: The unique identifier of the feature flag
+- feature_flag.provider_name: The name of the service provider that performs the flag evaluation
+- feature_flag.variant: SHOULD be a semantic identifier for a value. If one is unavailable, a stringified version of
+  the value can be used
+
+Given below is the list of metrics proposed and their usage,
+
+### feature_flag.evaluation_request_total
+
+A counter[2] based metric to calculate the number of flag evaluation requests. This is recorded at `before` stage of
+the hook.
+
+This metric is useful to understand flag evaluation requests received through OpenFeature SDK.
+
+### feature_flag.evaluation_success_total
+
+A counter[2] based metric to calculate the number of successful flag evaluations. This is recorded at `after` stage of
+the hook.
+
+This metric is useful to understand successful flag evaluations. This metric contain following extra dimension(s),
+
+- reason : evaluation reason extracted from flag resolution details[3]
+
+### feature_flag.evaluation_error_total
+
+A counter[2] based metric to calculate the number of failed flag evaluations. This is recorded at `error` stage of
+the hook.
+
+This metric is useful to understand flag evaluation errors. This metric contain following extra dimension(s),
+
+- exception : error/exception message extracted from the evaluation error
+
+### feature_flag.evaluation_active_count
+
+An UpDownCounter[4] based metric to calculate the number of active flag evaluations currently going through
+OpenFeature SDK. This is increased at `before` stage and decreased at `finally` stage.
+
+Given the evaluation is fast, the value observed here can be 0. However, when there are bottlenecks or provider
+slowdowns, this will be a non-zero value.
+
+## Expansion options
+
+Given below are future expansions that can be build on top of the metrics hook. These options will not be
+implemented as they require further discussions and agreements from the community.
+
+### Metric to measure latency
+
+credits - Justin Abrahms
+
+Flag evaluation latency can be calculated with time measurements between `finally` and `before` stages. However,
+this requires time measurement to be shared between two stages, which require either a context propagation or a shared
+variable (potentially a map). Alternatively, SDK could mark the evaluation start timestamp to enhance the accuracy
+of the measurement
+
+### Scope dimension
```

Comment:
This needs to be included in the OFEP. It would be optional and only applied if a provider supplies it as flag metadata. Without it, these metrics are significantly less useful with providers that support multiple environments.

Reply:
I will first update go-sdk implementation & then update the OFEP

Reply:
I added a new section on `extra dimension` [1], backed by go-sdk implementation to prove the feasibility [2]

[1] - https://github.com/open-feature/ofep/pull/62/files#diff-1ee0937abd11c52172d0be2b854ab5162b15262da53e7ddbcd4090484879bc67R61-R85
[2] - https://github.com/open-feature/go-sdk-contrib/pull/217